bigquery/google/cloud/bigquery/client.py

```diff
@@ -1121,6 +1196,14 @@ def copy_table(
 
         job_ref = job._JobReference(job_id, project=project, location=location)
 
+        if isinstance(sources, str):
+            sources = TableReference.from_string(
+                sources, default_project=self.project)
+
+        if isinstance(destination, str):
+            destination = TableReference.from_string(
+                destination, default_project=self.project)
```

Comment:
Nice!!
I've seen this implemented equivalent to `destination = maybe_coerce(destination)`, where `maybe_coerce` does the type checking; this is more explicit at the cost of the `if`s.

Reply:
That's a good idea. I wish I'd thought of that before I added all these `if`s. :-)